baybe/kernels/basic.py

```diff
@@ -43,3 +45,45 @@ class MaternKernel(Kernel):
 
     Only takes the values 0.5, 1.5 or 2.5. Larger values yield smoother interpolations.
     """
+
+    lengthscale_prior: Optional[Prior] = field(default=None)
+    """An optional prior on the kernel lengthscale."""
+
+    lengthscale_initial_value: Optional[float] = field(
+        default=None, converter=optional(float)
+    )
+    """An optional starting value for the kernel lengthscale."""
+
+    def to_gpytorch(self, *args, **kwargs):  # noqa: D102
+        # See base class.
+        from torch import Tensor
+
+        gpytorch_kernel = super().to_gpytorch(*args, **kwargs)
+        if self.lengthscale_initial_value is not None:
+            gpytorch_kernel.lengthscale = Tensor([self.lengthscale_initial_value])
+        return gpytorch_kernel
+
+
+@define(frozen=True)
+class ScaleKernel(Kernel):
+    """A kernel for decorating existing kernels with an outputscale."""
+
+    base_kernel: Kernel = field()
+    """The base kernel that is being decorated."""
+
+    outputscale_prior: Optional[Prior] = field(default=None)
+    """An optional prior on the output scale."""
+
+    outputscale_initial_value: Optional[float] = field(
+        default=None, converter=optional(float)
+    )
+    """An optional initial value for the output scale"""
+
+    def to_gpytorch(self, *args, **kwargs):  # noqa: D102
```

Comment:
this and the `to_gpytorch` method from Matern kenrel seem identical except the name of the `*_scale`

This means the `to_gpytorch` implementation could go into the base class (possibly uisng `hasattr` etc). This would reduce a lot the boilerplate code that would be expected now when adding other kernels

Reply:
This is exactly how it was implemented but then @AdrianSosic complained about the `hasattr` matching stuff being ugly and asking me to instead have this stuff here :D You can see the previous implementation some commits earlier. I'll let you discuss which variant you want and then implement.

Reply:
Here is the corresponding comment (copied and resolved the corresponding thread to avoid duplicate discussions):![alt text](image.png)

Reply:
ok I see
But conside rnow the situation that we might include 10 additional kernels with a `lengthscale`. They all have the adidtional boilerpalte `to_gpytorch` as we currently have for `MaternKernel`, imo not optimal

Solution 
1) Have an intermediate class providing the same `to_gpytorch` for all `lengthscale`-only kernels
2) Have the base kernel `to_gpytorch` simply handle `outputscale` and `lengthscale` via `hasattr` (which i guess was implemented before)

Reply:
Ok, I see your point with repeated code. However, let me suggest that we change this as soon as we actually introduce a new Kernel with lengthscale, since I see higher priority of merging the current PR asap. In the current version, there is no duplicate code yet and the change will not be user-facing, so we can easily adjust later when really needed. Also, I vote against Option 1 (not a big fan of all these intermediate classes that only add one property) perhaps rather for Option 2 but with a `ClassVar` instead of the rather type-unsafe `hasattr`. But let's postpone this the PR where we bring the next lengthscale kernel. Will resolve for now.

Reply:
can you pleaser drop the corresponding commit? doesnt seem reasonable to require the changes in the imminent more-kernels PR if the unwanted changes could simply be reversed here achieving the same

Reply:
Have an idea, let me draft a commit for it

Reply:
So how about the current version? Outputscale is still controlled in `ScaleKernel`, since it's the only kernel that has the attribute. Lengthscale, on the other hand, is controlled in the base class, but without hacky `hasattr` checks. Instead, we piggyback on the already existing attribute of the gpytorch base kernel class. Advantage: it's fail safe, i.e. we'll notice in case gpytorch backend code changes. If you like it, give it a thumbs up and I can resolve + we can proceed to merge

Reply:
That looks like a good solution at this point :)

Reply:
Ok, I had to silence a mypy warning (just amended the commit) but I think this is acceptable here. Added a comment that justifies the approach.

Reply:
agree that looks good :)

I dont like the lengthy comment tho
`justifications` for why this approach and not that approach belong in here, not in the code unless required for understanding (not the case here)

further, IF I was VERY picky I'd complain that the code setting the `lengtscale` now looks different now than the code setting the `outputscale` (one uses walruss one not) even tho they do the exact same thing

Reply:
pickiness appreciated. Will change the other code part as well (reason for the difference: one was written by alex, the other by me).

Regarding the comment: have to disagree here, tbh. If I put the justification here, no one will ever find it again when working on the code. IMO it's not immediately obvious why it is OK to ignore mypy instead of going with the (perhaps more immediate) approach of using `hasattr`. Keeping it in the code prevents people (including myself) to repeat the thinking process in a future refactoring.